### ISSUE TITLE
Clarify config options and transitively included features

### ISSUE THREAD

**Issue by Person-93** [frewsxcv/cargo-all-features#25]
### Current Config (as I understand it)
- `skip_feature_sets`: A set of sets. Any feature set from the matrix that is a superset of any members will be dropped.
- `skip_optional_dependencies`: Doesn't include optional dependencies when building the matrix.
- `extra_features`: a set of features that will be included in every set in the matrix.
- `allowlist`: If present, the matrix will only be built using these features. Other features may be transitively included.
- `denylist`: If present these features will not be used to build the matrix but others may be transitively enabled.

### Possible Sources of Confusion
- Perhaps `skip_feature_sets` can be renamed `conflicts` or `conflicting` for clarity.
- Default value for `skip_optional_dependencies` is `false` but the example in the README shows it as `true` and says in a comment that skipping them can improve performance.
- Perhaps `allowlist` should be renamed as it implies that any feature set that (directly or transitively) includes a feature that's not in the list will be dropped from the matrix. I'm not sure what a good name would be. The best I could come up with is `seed` because these values are the seeds from which the matrix is build.
- Perhaps `denylist` should be renamed to the opposite of whatever `allowlist` gets named for the same reason.
- Features that start with `__` are skipped but this is not documented and there is no way to disable it.

### Transitively Enabled Features
Currently, transitively enabled features are not taken into consideration when building the matrix. I think that they should be taken into account for a few reasons:

#### Smaller Matrix
More feature sets can be excluded from the matrix as duplicates, This would help with combinatorial explosion. Currently a crate with 10 features/optional-deps would have 1024 feature sets in its matrix. If some of those features enabled other features we can significantly cut down on testing time.

#### Better Restrictions

- We can allow users to deny transitively enabled features or optional dependencies. This would allow things like having a `__nightly` feature that is enabled by any other feature that uses nightly features and denying `__nightly`.
- If conflicting features get transitively enabled, they can be dropped from the matrix.

**Comment by demmerichs** [comment]
I agree, that transitively features should be enabled. I am working on a PR for this.

Regarding the naming of settings perhaps a new issue should be open to keep the topics separate, however, I find allowlist and denylist acceptable names and skip_optional_dependencies could be clarified in the docs.
